src/app/plugins/animal.plugin.ts

```diff
@@ -0,0 +1,195 @@
+import Constants from '../../common/constants';
+import { Plugin } from '../../common/plugin';
+import { ChannelType, IContainer, IHttpResponse, IMessage, Maybe } from '../../common/types';
+import { MessageEmbed } from 'discord.js';
+import ms from 'ms';
+
+export default class AnimalPlugin extends Plugin {
+  public commandName: string = 'animal';
+  public name: string = 'Animal Plugin';
+  public description: string = 'Get wild with a fraction of gods creatures';
+  public usage: string =
+    'animal <Species (optional)> <subSpecies(Optional)> | animal list (species or <species> or subSpecies)';
+  public pluginAlias = ['wild', 'species'];
+  public permission: ChannelType = ChannelType.Public;
+  public pluginChannelName: string = Constants.Channels.Public.Pets;
+
+  private _API_URL: string = 'https://api.fetchit.dev/';
+
+  private _validAnimals: Set<string> = new Set([]); // All animals and species
+  private _animals: ISubSpecies[] = []; // All species.
+
+  private _animalEmbed: Maybe<MessageEmbed>;
+  private _subspeciesEmbed: Maybe<MessageEmbed>;
+
+  private _ANIMAL_UPD_THRESH: number = ms('1d');
+  private _LAST_UPD_TIME: number = 0;
+
+  constructor(public container: IContainer) {
+    super();
+    this._updateAnimalAPIData();
+  }
+
+  public execute(message: IMessage, args?: string[]) {
+    this._updateAnimalAPIData();
+    const input = args?.length ? args : [this._getRandomAnimal()];
+
+    // Checks if they want to know what species/ subspecies are available.
+    if (input[0].startsWith('list')) {
```

Comment:
array destructuring

Reply:
Not sure what you getting at.  did you want the destructuring removing?

Reply:
use array destructuring for `input`